tubesync/sync/models.py

```diff
@@ -482,6 +489,15 @@ def type_directory_path(self):
     def make_directory(self):
         return os.makedirs(self.directory_path, exist_ok=True)
 
+    @property
+    def get_thumbnail_url(self):
+        if self.source_type==Source.SOURCE_TYPE_YOUTUBE_PLAYLIST:
+            raise Exception('This source is a playlist so it doesn\'t have thumbnail.')
+        soup = BeautifulSoup(requests.get(self.url, cookies={'CONSENT': 'YES+1'}).text, "html.parser")
+        data = re.search(r"var ytInitialData = ({.*});", str(soup.prettify())).group(1)
+        json_data = json.loads(data)
+        return json_data["header"]["c4TabbedHeaderRenderer"]["avatar"]["thumbnails"][2]["url"]
```

Comment:
This likely needs to be wrapped in a `KeyError` catch to politely handle YouTube changing their JSON format.

Reply:
This reminded me, I believe `yt-dlp` extracts the posters/banners for channels and adds them to the metadata for media items. There are usually very large images in the "thumbnails" metadata key with resolutions of 1920x1080. Are these the same as the posters fetched above? Could these be used instead?

Where possible it would be highly preferable to use `yt-dlp`'s well maintained upstream for features.